rewrite-core/src/main/java/org/openrewrite/FindSourceFiles.java

```diff
@@ -51,15 +52,21 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
             public Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
                 if (tree instanceof SourceFile) {
                     SourceFile sourceFile = (SourceFile) tree;
-                    String sourcePath = sourceFile.getSourcePath().toString();
-                    if (StringUtils.matchesGlob(sourcePath.matches("^\\.?[/\\\\]") ? sourcePath :
-                            "./" + sourcePath, filePattern)) {
-                        results.insertRow(ctx, new SourcesFiles.Row(sourcePath));
+                    Path sourcePath = sourceFile.getSourcePath();
+                    if (PathUtils.matchesGlob(sourcePath, normalize(filePattern))) {
+                        results.insertRow(ctx, new SourcesFiles.Row(sourcePath.toString()));
                         return SearchResult.found(sourceFile);
                     }
                 }
                 return tree;
             }
         };
     }
+
+    private static String normalize(String filePattern) {
+        while (filePattern.startsWith(".") || filePattern.startsWith("/") || filePattern.startsWith("\\")) {
```

Comment:
I think we will slightly have to change this. Please note the example given in the documentation of `FindSourceFiles#filePattern`: `.github/workflows/*.yml`. Using this code, the normalization would cut off the leading `.`.

Would be cool if we could also add a test case for this to `FindSourceFilesTest` (since this logic is part of the recipe).

Reply:
Right, right. I forgot to handle the cases for dotfiles. I'll fix that up and add some test cases to ensure those don't break.

One aspect that I kept going back and forth on is if `PathUtils#matchesGlob` should consider the current directory as part of it's matching logic. For now, I opted not to but it's still a thought in my head.

Ex:
```
hello.txt != ./hello.txt
```

This is effectively what the normalization is trying to handle anyway.

Reply:
Fixed dotfiles via [d9c3c07](https://github.com/openrewrite/rewrite/pull/3067/commits/d9c3c077aa3e5e0d37b1066441010edf7a86c0f8).